pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java

```diff
@@ -57,7 +57,7 @@ public String getServerVersion() {
     public int getServerVersionNum() {
         if (serverVersionNum != 0)
             return serverVersionNum;
-        return Utils.parseServerVersionStr(serverVersion);
+        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
```

Comment:
Yes we don't seem to have much consistency with respect to thread safety.

I think in general though connections are not meant to be shared between
threads.

Postgresql itself is not particularly friendly with respect to that

Dave Cramer

On 27 December 2015 at 16:29, Valentine Gogichashvili <
notifications@github.com> wrote:

> In pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java
> <https://github.com/pgjdbc/pgjdbc/pull/464#discussion_r48458442>:
>
> > @@ -57,7 +57,7 @@ public String getServerVersion() {
> >      public int getServerVersionNum() {
> >          if (serverVersionNum != 0)
> >              return serverVersionNum;
> > -        return Utils.parseServerVersionStr(serverVersion);
> > +        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
>
> When looking into the code of this class, it looks like quite inconsistent
> about synchronization.
>
> Half of the methods are not synchronized though they access non-volatile
> inner variables.
>
> It definitely does not relate to that pull request.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/464/files#r48458442>.
>